### PR TITLE
feat: setup logic for `resign` & `offer draw`

### PR DESCRIPTION
**Pull request by devsargam**

### CODE REVIEW

**Comment by hkirat** [comment]
nice

**Comment by Sama-004** [comment]
I tried implementing this but failed :(

**Comment by devsargam** [comment]
Hey @hkirat, could you please review this one?

File: apps/ws/src/GameManager.ts [hunk 1]
```diff
@@ -171,6 +172,44 @@ export class GameManager {
...
+            payload: {
+              from: message.payload.myColor,
+              id: message.payload.myId,
+            },
+          }),
+        );
+      }
+
+      if (message.type === DRAW_OFFER_ACCEPTED) {
+        SocketManager.getInstance().broadcast(
```
Comment (hkirat) [inline apps/ws/src/GameManager.ts:203]: Are we checking if the other user has offered a draw before accepting it?
It should be stored somewhere that  there was an offer made

Comment (devsargam) [reply apps/ws/src/GameManager.ts:203]: good one. I thought better do this way since doing db was a lot of hassle. Happy to change if needed? is it?

Comment (devsargam) [reply apps/ws/src/GameManager.ts:203]: @hkirat how about a in memory store the expires every 30s?

Comment (hkirat) [reply apps/ws/src/GameManager.ts:203]: In memory yeah makes sense

File: apps/ws/src/GameManager.ts [hunk 2]
```diff
@@ -171,6 +173,56 @@ export class GameManager {
...
+            payload: {
+              from: message.payload.myColor,
+              id: message.payload.myId,
+            },
+          }),
+        );
+      }
+
+      if (message.type === DRAW_OFFER_ACCEPTED) {
+        const drawOfferCache = Cache.getInstance().get('draw_offer', [
```
Comment (hkirat) [inline apps/ws/src/GameManager.ts:211]: Remove cache logic for now

**Comment by devsargam** [comment]
https://www.loom.com/share/3a6f9bcefc164fe2b157ca3850b34ac0

**Comment by devsargam** [comment]
image

**Comment by devsargam** [comment]
Hey @hkirat kindly check this one. A lot of ui changes very prone to merge conflicts

**Comment by devsargam** [comment]
Ui can be extended from #223

**Comment by hkirat** [comment]
https://github.com/code100x/chess/pull/223 is merged
Do u want to bring this one up to date or create a new one @devsargam ?

**Comment by devsargam** [comment]
@hkirat new one should do it. Raising it by eod

**Pull request closed without merging**